src/Components/Components/src/LayoutView.cs

```diff
@@ -0,0 +1,77 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Reflection;
+using System.Threading.Tasks;
+
+namespace Microsoft.AspNetCore.Components
+{
+    /// <summary>
+    /// Displays the specified content inside the specified layout and any further
+    /// nested layouts.
+    /// </summary>
+    public class LayoutView : IComponent
+    {
+        private static RenderFragment EmptyRenderFragment = builder => { };
+
+        private RenderHandle _renderHandle;
+
+        /// <summary>
+        /// Gets or sets the content to display.
+        /// </summary>
+        [Parameter]
+        public RenderFragment ChildContent { get; set; }
+
+        /// <summary>
+        /// Gets or sets the type of the layout in which to display the content.
+        /// The type must implement <see cref="IComponent"/> and accept a parameter named <see cref="LayoutComponentBase.Body"/>.
+        /// </summary>
+        [Parameter]
+        public Type Layout { get; set; }
+
+        /// <inheritdoc />
+        public void Attach(RenderHandle renderHandle)
```

Comment:
Is there a good reason why not to inherit component base here? We don't want to handle events?

Reply:
In general for built-in components, I like going directly to `IComponent` unless we're getting some real value from the `ComponentBase` lifecycle. Benefits include:

 * The component will do less stuff at runtime (it's not running through a set of lifecycle stages that do nothing)
 * It's easier to reason about for unit tests
 * It cements the fact that the underlying contract is `IComponent` and we don't start making assumptions (in the renderer, say) that all components are actually `ComponentBase`

I wouldn't be strongly against using `ComponentBase` for built-in components when it really saved us some complexity, but it doesn't in this case.